api/v1alpha1/types.go

```diff
@@ -47,3 +48,12 @@ func (m MeshGatewayConfig) toConsul() capi.MeshGatewayConfig {
 		}
 	}
 }
+
+func (m MeshGatewayConfig) validate() *field.Error {
+	switch m.Mode {
+	case "", "local", "remote", "none":
```

Comment:
Can we make this a `var meshGatewayModes = []string{"", "local", "remote", "none"}` and then use it for this case and constructing the error message?